kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/FilterFactory.java

```diff
@@ -9,16 +9,9 @@
 
 /**
  * FilterFactory is a pluggable source of {@link Filter} instances.
- * @param <F> the {@code Filter} type.
  * @param <C> the type of configuration used to create the {@code Filter}. Use {@link Void} if the {@code Filter} is not configurable.
  */
-public interface FilterFactory<F extends Filter, C> {
-
-    /**
-     * @return The concrete class of {@code Filter} this factory {@linkplain #createFilter(FilterCreationContext, C) creates}.
-     */
-    @NonNull
-    Class<F> filterType();
+public interface FilterFactory<C> {
```

Comment:
It's not really clear whether we should parameterise the factory with the config type. The `configType()` method doesn't really provide much value to the runtime. It's not a pattern I've repeated in all the other plugins, but we should probably apply some pattern consistently.